File: python/channels/split-channel.py
#!/usr/bin/python env

import cv2 as cv
import numpy as np
from matplotlib import pyplot as plt
import argparse

# adding parsing arguments
parser = argparse.ArgumentParser(description='Splits rgb into red, green and blue channel')
parser.add_argument("--image", type=str, help="path to image file")
args = vars(parser.parse_args())

# read this file
imgToLoad = args["image"]
fig, axes = plt.subplots(1, 4)
fig.tight_layout()
fig.suptitle("Splitting Image into Red, Green & Blue")

# img to load
imgRgb = cv.cvtColor(cv.imread(imgToLoad), cv.COLOR_BGR2RGB)

# split channel using cv.split
(r, g, b) = cv.split(imgRgb)

imgs = [imgRgb, r, g, b]
headings = ["Orignal Image", "Red Channel", "Green Channel", "Blue Channel"]

for i in range(0, len(imgs)):
    axes[i].axis("off")
    axes[i].set_title(headings[i])
    axes[i].imshow(imgs[i], cmap='gray')
    # uncomment if wanna save the indivvual files 
    # cv.imwrite(f"dataset/{headings[i]}.jpg", imgs[i] )

plt.show()

File: matlab/src/split_channel.m
% split channels into rgb

clc;
rows  = 1; 
cols = 4;
% reading an image from ../dataset/
I = imread("../dataset/wall.jpg");

% subplotting original image

subplot(rows, cols, 1), imshow(I);
title("Original");

% extracting rgb channels
redChannel = I(:,:,1); 
greenChannel = I(:,:,2);
blueChannel = I(:,:,3);

% plotting red, green and blue channel
subplot(rows, cols, 2), imshow(redChannel);
title('Red Channel');
subplot(rows, cols, 3), imshow(greenChannel);
title('Green Channel');
subplot(rows, cols, 4), imshow(blueChannel);
title('Blue Channel');